SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/phosphorus_mineralization.py

```diff
@@ -0,0 +1,341 @@
+        indicates that there is more labile phosphorus than there should be, and a return value of zero indicates that
+        the two pools are balanced.
+
+        """
+        return labile_phosphorus - (active_phosphorus * (sorption_parameter / (1 - sorption_parameter)))
+
+    @staticmethod
+    def _calculate_phosphorus_desorption(active_inorganic_unbalanced_counter: int, sorption_parameter: float,
+                                         phosphorus_balance: float) -> float:
+        """Calculates how much phosphorus should be desorped in the given soil layer.
+
+        Parameters
+        ----------
+        active_inorganic_unbalanced_counter : int
+            The number of days that the active inorganic phosphorus pool has been greater than it would be when in
+            equilibrium with the labile inorganic phosphorus pool
+        sorption_parameter : float
+            The mean phosphorus sorption parameter that has been adjusted for the current day's amount of labile
+            inorganic phosphorus present (unitless)
+        phosphorus_balance : float
+            A value indicating how unbalanced the labile and active inorganic phosphorus pools are (unitless)
+
+        Returns
+        -------
+        float
+            The amount of phosphorus that should be removed from the active inorganic phosphorus pool to put it in
+            equilibrium with the labile inorganic phosphorus pool (kg phosphorus / ha)
+
+        References
+        ----------
+        SurPhos pminrl.f, lines 69, 70
+        Vadas P.A., Krogstad T., Sharpley A.N. (2006) Modeling phosphorus transfer between labile and nonlabile soil
+            pools: Updating the EPIC model. Soil Science Society of America Journal 70:736-743. DOI:
+            Doi 10.2136/Sssaj2005.0067. (Eqn. [8])
+
+        """
+        base = PhosphorusMineralization._determine_desorption_base(sorption_parameter)
+        desorption_factor = base * (active_inorganic_unbalanced_counter ** -0.32)
+        amount_transferred = desorption_factor * phosphorus_balance * -1
+        return amount_transferred
+
+    @staticmethod
+    def _determine_desorption_base(sorption_parameter: float) -> float:
+        """This method calculates a value used to determine how much phosphorus is desorped from the active inorganic
+            phosphorus pool in the labile organic phosphorus pool.
+
+        Parameters
+        ----------
+        sorption_parameter : float
+            The mean phosphorus sorption parameter that has been adjusted for the current day's amount of labile
+            inorganic phosphorus present (unitless)
+
+        Returns
+        -------
+        float
+            A value (named 'base' in code and literature) that is used to determine how much phosphorus is transferred
+            from the active inorganic phosphorus pool to the labile inorganic phosphorus pool.
```

Comment:
```suggestion
            from the active inorganic phosphorus pool to the labile inorganic phosphorus pool. (unitless)
```
I'm guessing unitless?

Reply:
Yes